### ISSUE TITLE
Dev branch: __WEBPACK_IMPORTED_MODULE_3_redux_saga___default(...) is not a function 

### ISSUE THREAD

**Issue by jwyuen** [mxstbr/react-boilerplate#950]
## Description

I'm getting this error in the console and nothing appears on the webpage:
`__WEBPACK_IMPORTED_MODULE_3_redux_saga___default(...) is not a function`

## Steps to reproduce

1) git clone https://github.com/mxstbr/react-boilerplate.git
2) git checkout origin/dev
3) npm run setup
4) npm start

# Versions

- Node/NPM: 6.5 / 3.10.3
- Browser: Chrome 51.0.2704 Kubuntu 16.04

**Comment by jwyuen** [comment]
Ok I figured out I could get a valid stack trace by editing the webpack.base.babel.js file and changing `devtool: options.devtool` to `devtool: '#inline-source-map'.  I then tracked the problem down to this statement in store.js:

`const sagaMiddleware = createSagaMiddleware();`

I did a `console.log(createSagaMiddleware)` and figured out it was no longer returning a function but instead an object.  This might point to this being a problem with redux-saga but I'm not quite sure...

In any case, changing the line in store.js to the following seems to solve the issue:

`const sagaMiddleware = createSagaMiddleware.default();`

**Comment by jwyuen** [comment]
Ok what's really curious is that if I clone the version of the boilerplate like so:
`git clone --depth=1 https://github.com/mxstbr/react-boilerplate.git`,

then change the version of redux-saga to 0.11.1 instead of 0.11.0, this problem doesn't crop up.  That would seem to indicate there's something else going on here....maybe related to webpack and/or babel?

**Comment by jwyuen** [comment]
I figured out if I change webpack.base.babel.js mainFields from:

```
mainFields: [
      'jsnext:main',
      'main',
    ],
```

To:

```
mainFields: [
      'main',
      'jsnext:main',
    ],
```

... the original code works.  I don't really understand why.  I figured this out while I was troubleshooting another issue with redux-form which pointed me to this solution:

https://github.com/mxstbr/react-boilerplate/issues/868

**Comment by jwyuen** [comment]
Holy moly ok so I think the real fix here is to update webpack from 2.1.0-beta.15 to 2.1.0-beta.21.  That seems to fix the root issue........weeeeeeee!

**Comment by gihrig** [comment]
@jwyuen Thanks for all the hard work!

I do the weekly dependency update. I have been holding at webpack-2.1.0-beta.15 because later versions cause problems see #841.

This weekend, I'll try an update to webpack-2.1.0-beta.21.

**Comment by gihrig** [comment]
I've been working on this, it goes pretty deep, I've run out of time :-/ Any input welcome!

Node 6.4.0 
npm 3.10.6 
Mac OS 10.11.6
Chrome 53.0.2785.89 (64-bit)

Here's what I have so far:

Ref #952 for PR
Ref #948 run up to 3.2.0

Error:
`__WEBPACK_IMPORTED_MODULE_3_redux_saga___default(...) is not a function`

Triggered by:
1. clean and start
2. in example app, access to any route

Update to `webpack@2.1.0-beta.21` remove node_modules and npm install

fixes 2.

Causes error in production build process:

`~/react-bp/node_modules/webpack/lib/Chunk.js:34
		throw new Error("Chunk.entry was removed. Use hasRuntime()");`

Fixed by extract-text-webpack-plugin 1.0.1  →  2.0.0-beta

See [webpack #2764 Comment](https://github.com/webpack/webpack/issues/2764#issuecomment-232371501)

Causes error in production build process:

`Error: Breaking change: extract now only takes a single argument. Either an options object *or* the loader(s)`.
See: https://github.com/webpack/extract-text-webpack-plugin#usage-example-with-css

Fixed by modifying `webpack.prod.babel.js:27+` to use new ExtractTextPlugin syntax

Causes error in production example app

Many styles are missing, e.g. `Home` and `Features` links are not formatted as buttons

**Comment by gihrig** [comment]
Problem fixed. See PR #952.
